src/components/ProgressBar/ProgressBar.jsx

```diff
@@ -0,0 +1,91 @@
+import PropTypes from 'prop-types'
+import React, { useMemo } from 'react'
+import styled, { css } from '@xstyled/styled-components'
+
+import { Flex, Caption } from '../'
+
+import SvgCheckCircle from '../../icons/CheckCircle'
+import SvgError from '../../icons/Error'
+
+const ProgressBar = ({ size, type, successColor, loadingColor, errorColor, progress, error }) => {
+  const barColor = useMemo(() => {
+    if (progress === 100) {
+      return successColor
+    }
+
+    if (error === true) {
+      return errorColor
+    }
+
+    return loadingColor
+  }, [error, progress])
+
+  return (
+    <Flex alignItems='center'>
+      <BackgroundBar size={size}>
+        <InsideBar progress={progress} backgroundColor={barColor} />
+      </BackgroundBar>
+
+      {type === 'withPercent' && (
+        <Caption fontSize='12px' marginLeft='5px' color='#757575'>
+          {progress}%
+        </Caption>
+      )}
+
+      {progress === 100 && type === 'withIcon' && (
+        <Flex ml='5px'>
+          <SvgCheckCircle ml='5px' fill={successColor} />
+        </Flex>
+      )}
+
+      {error === true && type === 'withIcon' && (
```

Comment:
não precisa usar `=== true` :)

```suggestion
      {error && type === 'withIcon' && (
```